homeassistant/components/climate/__init__.py

```diff
@@ -235,10 +235,17 @@ def aux_heat_set_service(service):
     def temperature_set_service(service):
         """Set temperature on the target climate devices."""
         target_climate = component.extract_from_service(service)
-        kwargs = service.data
+
         for climate in target_climate:
-            climate.set_temperature(**kwargs)
+            kwargs = {}
+            for value, temp in service.data.items():
+                kwargs[value] = convert_temperature(
```

Comment:
Just tested this with the demo component setting an F device, when I have my config set to C, so the convert function will actually do something. Problem confirmed:
```
16-10-02 13:08:40 INFO (MainThread) [homeassistant.core] Bus:Handling <Event call_service[L]: service=set_temperature, service_call_id=140337833522792-2, service_data=temperature=40, entity_id=climate.heatpump, domain=climate>
16-10-02 13:08:40 ERROR (ThreadPool Worker 8) [homeassistant.core] BusHandler:Exception doing job
Traceback (most recent call last):
  File "/home/martin/Dev/home-assistant/homeassistant/core.py", line 1167, in job_handler
    func(*args)
  File "/home/martin/Dev/home-assistant/homeassistant/core.py", line 1027, in execute_service
    service_handler.func(service_call)
  File "/home/martin/Dev/home-assistant/homeassistant/components/climate/__init__.py", line 245, in temperature_set_service
    climate.unit_of_measurement
  File "/home/martin/Dev/home-assistant/homeassistant/util/temperature.py", line 32, in convert
    return celsius_to_fahrenheit(temperature)
  File "/home/martin/Dev/home-assistant/homeassistant/util/temperature.py", line 17, in celsius_to_fahrenheit
    return celsius * 1.8 + 32.0
TypeError: can't multiply sequence by non-int of type 'float'
```